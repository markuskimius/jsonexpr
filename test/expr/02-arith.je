PRINT("== STRING ==");
PRINT("Hello, world!");
PRINT("Hello, " + 3.141592);
PRINT("Hello, " + (3.141592 * 2));
PRINT();

PRINT("== OPERATOR PRECEDENCE ==");
PRINT(2 + 3 + 4 + 5);
PRINT(2 - 3 + 4 - 5);
PRINT(2 - 3 - 4 - 5);
PRINT(2 - 3 * 4 + 5);
PRINT(2 - 3 * (4 + 5));
PRINT((2 - 3) * 4 + 5);
PRINT();

PRINT(+2 + +3 + +4 + +5);
PRINT(+2 - +3 + +4 - +5);
PRINT(+2 - +3 - +4 - +5);
PRINT(+2 - +3 * +4 + +5);
PRINT(+2 - +3 * (+4 + +5));
PRINT((+2 - +3) * +4 + +5);
PRINT();

PRINT(-2 + -3 + -4 + -5);
PRINT(-2 - -3 + -4 - -5);
PRINT(-2 - -3 - -4 - -5);
PRINT(-2 - -3 * -4 + -5);
PRINT(-2 - -3 * (-4 + -5));
PRINT((-2 - -3) * -4 + -5);
PRINT();

PRINT(2.5 + 3.5 + 4.5 + 5);
PRINT(2.5 - 3.5 + 4.5 - 5);
PRINT(2.5 - 3.5 - 4.5 - 5);
PRINT(2.5 - 3.5 * 4.5 + 5);
PRINT(2.5 - 3.5 * (4.5 + 5));
PRINT((2.5 - 3) * 4.5 + 5);
PRINT();

PRINT(+2.5 + +3.5 + +4.5 + +5);
PRINT(+2.5 - +3.5 + +4.5 - +5);
PRINT(+2.5 - +3.5 - +4.5 - +5);
PRINT(+2.5 - +3.5 * +4.5 + +5);
PRINT(+2.5 - +3.5 * (+4.5 + +5));
PRINT((+2.5 - +3) * +4.5 + +5);
PRINT();

PRINT(-2.5 + -3.5 + -4.5 + -5);
PRINT(-2.5 - -3.5 + -4.5 - -5);
PRINT(-2.5 - -3.5 - -4.5 - -5);
PRINT(-2.5 - -3.5 * -4.5 + -5);
PRINT(-2.5 - -3.5 * (-4.5 + -5));
PRINT((-2.5 - -3) * -4.5 + -5);
PRINT();

PRINT(33 + 17 - 8 + 3);
PRINT(33 + 17 * 8 + 3);
PRINT(33 + 17 / 8 + 3);
PRINT(33 + 17 % 8 + 3);
PRINT(33 * 17 / 8 * 3);
PRINT(33 * 17 % 8 * 3);
PRINT(33 / 17 % 8 / 3);
PRINT(33.0 + 17.0 - 8.0 + 3.0);
PRINT(33.0 + 17.0 * 8.0 + 3.0);
PRINT(33.0 + 17.0 / 8.0 + 3.0);
PRINT(33.0 + 17.0 % 8.0 + 3.0);
PRINT(33.0 * 17.0 / 8.0 * 3.0);
PRINT(33.0 * 17.0 % 8.0 * 3.0);
PRINT(33.0 / 17.0 % 8.0 / 3.0);
PRINT();

PRINT("== INCREMENT ==");
i=4; PRINT(2 - 3 - i++ - 5);
i=4; PRINT(2 - 3 - ++i - 5);
i=4.0; PRINT(2 - 3 - i++ - 5);
i=4.0; PRINT(2 - 3 - ++i - 5);
PRINT();

PRINT("== DECREMENT ==");
i=4; PRINT(2 - 3 - i-- - 5);
i=4; PRINT(2 - 3 - --i - 5);
i=4.0; PRINT(2 - 3 - i-- - 5);
i=4.0; PRINT(2 - 3 - --i - 5);
PRINT();

PRINT("== POWER ==");
PRINT(7 ** 3);
PRINT(7 ** -3);
PRINT(-7 ** 3);
PRINT(-7 ** -3);
PRINT();

PRINT(7 ** 0.5);
PRINT(7 ** -0.5);
PRINT(-7 ** 0.5);
PRINT(-7 ** -0.5);
PRINT();

PRINT(7.5 ** 3);
PRINT(7.5 ** -3);
PRINT(-7.5 ** 3);
PRINT(-7.5 ** -3);
PRINT();

PRINT(7.5 ** 3.5);
PRINT(7.5 ** -3.5);
PRINT(-7.5 ** 3.5);
PRINT(-7.5 ** -3.5);
PRINT(7.5 ** 0.5);
PRINT(7.5 ** -0.5);
PRINT(-7.5 ** 0.5);
PRINT(-7.5 ** -0.5);
