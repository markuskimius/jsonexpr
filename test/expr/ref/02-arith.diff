+0.37796447300922726
-0.37796447300922721
+1155.3522697374596
-1155.3522697374598
