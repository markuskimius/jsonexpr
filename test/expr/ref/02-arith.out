== STRING ==
Hello, world!
Hello, 3.1415920000000001
Hello, 6.2831840000000003

== OPERATOR PRECEDENCE ==
14
-2
-10
-5
-25
1

14
-2
-10
-5
-25
1

-14
2
10
-19
-29
-9

15.5
-1.5
-10.5
-8.25
-30.75
2.75

15.5
-1.5
-10.5
-8.25
-30.75
2.75

-15.5
1.5
10.5
-23.25
-35.75
-7.25

45
172
38
37
210
3
0
45.0
172.0
38.125
null
210.375
null
null

== INCREMENT ==
-10
-11
-10.0
-11.0

== DECREMENT ==
-10
-9
-10.0
-9.0

== POWER ==
343.0
0.0029154518950437315
-343.0
-0.0029154518950437315

2.6457513110645907
0.37796447300922721
nan
nan

421.875
0.0023703703703703702
-421.875
-0.0023703703703703702

1155.3522697374598
0.00086553688099581798
nan
nan
2.7386127875258305
0.36514837167011071
nan
nan
